plugins/module_utils/syslog_server.py

```diff
@@ -144,6 +146,9 @@ def get_state(
             )
         ]
 
+        for s in state:
+            s["protocol"] = protocols[s["protocol"]]  # type: ignore
+
```

Comment:
Please move this part of code into `from_hypercore()` method - this is the place to do mapping from hypercore representation into our internal representation.

Reply:
You shouldn't feel bad about it, that part of code was actually not good for my eyes either and knew there should be a better way to go about it, I just couldn't think of it at the moment. :)